src/main/java/view/ClassConfig.java

```diff
@@ -38,7 +38,7 @@ public ClassConfig(String courseName, String semester, JPanel curPanel){
         btnAdd.addActionListener(new ActionListener() {
 
             public void actionPerformed(ActionEvent e) {
-                NewCategoryDialog dialog = new NewCategoryDialog((JFrame) SwingUtilities.getWindowAncestor(curPanel));
+                NewCategoryDialog dialog = new NewCategoryDialog((JFrame) SwingUtilities.getWindowAncestor(thisComponent));
```

Comment:
add this line seems better:
JPanel curPanel = (JPanel) ((JButton) e.getSource()).getParent();